crates/svm-runtime/tests/runtime_tests.rs

```diff
@@ -325,3 +325,61 @@ fn runtime_exec_app_reaches_oog() {
 
     assert_eq!(expected, actual)
 }
+
+#[test]
+fn default_runtime_func_buf() {
+    let mut runtime = default_runtime!();
+
+    // 1) deploying the template
+    let version = 0;
+    let author = Address::of("author").into();
+    let is_wast = true;
+    let maybe_gas = MaybeGas::new();
+    let layout: DataLayout = vec![8].into();
+
+    let bytes = testing::build_template(
+        version,
+        "My Template",
+        layout.clone(),
+        include_str!("wasm/runtime_func_buf.wast"),
+        is_wast,
+    );
+
+    let receipt = runtime.deploy_template(&bytes, &author, HostCtx::new(), maybe_gas);
+    assert!(receipt.success);
+
+    let template_addr = receipt.addr.unwrap();
+
+    // 2) spawn app
+    let ctor_idx = 0;
+    let ctor_buf = vec![];
+    let ctor_args = vec![];
+    let creator = Address::of("creator").into();
+
+    let bytes = testing::build_app(version, &template_addr, ctor_idx, &ctor_buf, &ctor_args);
+    let receipt = runtime.spawn_app(&bytes, &creator, HostCtx::new(), maybe_gas);
+    assert!(receipt.success);
+
+    let app_addr = receipt.get_app_addr();
+    let init_state = receipt.get_init_state();
+
+    // 3) executing an app-transaction
+    let func_idx = 1;
+    let func_buf = vec![0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 0x80];
+    let func_buf_ptr = WasmValue::I32(0);
```

Comment:
Can you give more explanations how this works (in conjunction with the `runtime_func_buf.wast` code)?

Reply:
I have an idea of how to make it easier to understand.

I prefer to open a new PR that will add two new vmcalls (that we'll need)
and change the test to be easier